taskdb/src/redis_db.rs

```diff
@@ -140,10 +153,18 @@ impl RedisTaskDb {
     fn new(config: RedisConfig) -> RedisDbResult<Self> {
         let url = config.url.clone();
         let client = Client::open(url).map_err(RedisDbError::RedisDb)?;
-        let conn = client.get_connection().map_err(RedisDbError::RedisDb)?;
-        Ok(RedisTaskDb { conn, config })
+        Ok(RedisTaskDb { client, config })
     }
 
+    fn get_conn(&mut self) -> Result<Connection, RedisError> {
+        match self.client.get_connection() {
+            Ok(conn) => Ok(conn),
+            Err(_) => {
+                self.client = redis::Client::open(self.config.url.clone())?;
+                self.client.get_connection()
```

Comment:
```suggestion
                self.get_connection()
```

Reply:
you mean recursive call?

Reply:
Yes. BTW, I would suggest retrying multiple times with a backoff time interval, because the Redis restart may take some time.

Reply:
recursive call is kind of risky for infinite loop, backoff time interval with max retry probably be better. update soon.

Reply:
Done!

Reply:
LGTM